### PR TITLE
Updating recommended elasticsearch version

### LINKED ISSUE #404: Operate: Update Elasticsearch dependency version
We declare that Operate supports Elasticsearch 7.12 [here](https://docs.camunda.io/docs/components/operate/deployment/install-and-start/#download-elasticsearch ).

Need to be updated to 7.16.1

### PR DESCRIPTION
**Pull request by santanaluiz**
Updating recommended elasticsearch version

NOTE: link says 404 but it is the link provided in here:
https://www.elastic.co/downloads/past-releases#elasticsearch
I am assuming they will fix it soon

Closes #404

### CODE REVIEW

File: versioned_docs/version-1.0/components/tasklist/deployment/install-and-start.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ On the Zeebe release page, compatible versions of Zeebe and Tasklist are grouped
 
 Tasklist uses open-source Elasticsearch as its underlying data store, and so to run Tasklist, you need to download and run Elasticsearch.
 
-Tasklist is currently compatible to Elasticsearch 7.12.1 [You can download Elasticsearch here.](https://www.elastic.co/downloads/past-releases/elasticsearch-7-12-1)
+Tasklist is currently compatible with Elasticsearch 7.16.1. Download Elasticsearch [here](https://www.elastic.co/downloads/past-releases/elasticsearch-7-16-1).
```
Comment (menski) [inline versioned_docs/version-1.0/components/tasklist/deployment/install-and-start.md:40]: as 1.0 is not supported anymore, I guess we don't have to update it here, wdyt?

**Review by menski — CHANGES_REQUESTED** [review]
I guess we should also update the versions on https://docs.camunda.io/docs/reference/supported-environments/

**Review by christinaausley — APPROVED** [review]
Really appreciate you shifting the links to [here] rather than several words -- from a technical writing perspective this looks good to go.

**Review by menski — APPROVED** [review]

File: versioned_docs/version-1.0/components/operate/deployment/install-and-start.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ On the Zeebe release page, compatible versions of Zeebe and Operate are grouped
 
 Operate uses open-source Elasticsearch as its underlying data store, and so to run Operate, you need to download and run Elasticsearch.
 
-Operate is currently compatible to Elasticsearch 7.12.1 [You can download Elasticsearch here.](https://www.elastic.co/downloads/past-releases/elasticsearch-7-12-1)
+Operate is currently compatible with Elasticsearch 7.16.1. Download Elasticsearch [here](https://www.elastic.co/downloads/past-releases/elasticsearch-7-16-1).
```
[the remark this answers is not in the document]
Comment (menski) [reply versioned_docs/version-1.0/components/operate/deployment/install-and-start.md]: I removed this change as we didn't updated 1.0 releases

File: versioned_docs/version-1.0/components/tasklist/deployment/install-and-start.md (hunk 1 above)
Comment (menski) [reply versioned_docs/version-1.0/components/tasklist/deployment/install-and-start.md]: I removed this change as we didn't updated 1.0 releases

**Pull request merged**